src/System.Buffers.Primitives/System/Internal/ManagedBufferPool.cs

```diff
@@ -70,7 +80,17 @@ public override BufferHandle Pin(int index = 0)
                 }
             }
 
-            byte[] _array;
+            public override void Retain()
+            {
+                if (IsDisposed) BufferPrimitivesThrowHelper.ThrowObjectDisposedException(nameof(ArrayPoolBuffer));
+                Interlocked.Increment(ref _referenceCount);
+            }
+
+            public override void Release()
+            {
+                Debug.Assert(!IsDisposed);
```

Comment:
Should this be a Debug.Assert? What is preventing the caller to call Release multiple times?